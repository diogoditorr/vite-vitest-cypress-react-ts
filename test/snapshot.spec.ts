import { expect, it, test } from "vitest";

function toUpperCase(word: string) {
    return word.toUpperCase();
}

it("toUpperCase expect.toMatchSnapshot", () => {
    const result = toUpperCase("foobar");
    expect(result).toMatchSnapshot();
});

it("toUpperCase expect.toMatchInlineSnapshot", () => {
    const result = toUpperCase("foobar");
    expect(result).toMatchInlineSnapshot('"FOOBAR"');
});

test("snapshot object with match inline", () => {
    const bar = [
        {
            foo: "bar",
        },
    ];

    expect(bar).toMatchInlineSnapshot(`
    [
      {
        "foo": "bar",
      },
    ]
  `);
});

test("toThrowErrorMatchingSnapshot", () => {
    expect(() => {
        throw new Error("error");
    }).toThrowErrorMatchingSnapshot("hint");
});
