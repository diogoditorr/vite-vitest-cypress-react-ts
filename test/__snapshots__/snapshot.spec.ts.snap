// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toThrowErrorMatchingSnapshot > hint 1`] = `"error"`;

exports[`toUpperCase expect.toMatchSnapshot 1`] = `"FOOBAR"`;
